rfcs/20200205-standalone-keras-repository.md

```diff
@@ -0,0 +1,394 @@
+# Standalone Keras Repository
+
+| Status        | Proposed |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | [202](https://github.com/tensorflow/community/pull/202) |
+| **Author(s)** | Qianli Zhu (scottzhu@google.com), Francois Chollet (fchollet@google.com) |
+| **Sponsor**   | Karmel Allison (karmel@google.com) |
+| **Updated**   | 2020-02-05                         |
+
+## Objective
+
+Move the Keras code from the TensorFlow main GitHub repository to its own
+repository, with TensorFlow as a dependency.
+
+## Motivation
+
+### Build times
+
+Building the open-source TensorFlow project end-to-end is an extensive exercise. 
+With a standard GCP instance, it might take more than one hour to finish the whole
+build process (it might take longer with a Mac laptop). Although the local build 
+cache might help speed up the follow-up builds, the initial time cost is too 
+high for regular software development workflows. Internally, Google has a
+distributed build and caching service, which Googlers heavily rely on,
+that can build TensorFlow and run all Keras tests within 5 mins. Sadly,
+we can't expose this to external contributors.
+
+Currently, any contribution to Keras code will require building all of
+TensorFlow, which is quite expensive to do for average users.
+Having a separate repository will allow the Keras package to be built
+without building TensorFlow. This should greatly improve the 
+velocity of open-source developers when they contribute to Keras code.
+
+### Community Benefit
+
+The difficulty of building TensorFlow from scratch in order to make a PR
+to Keras code has been a significant source of issues:
+
+* It discouraged contributions, since many external developers couldn't test
+their changes and make sure they were correct.
+* External developers would send unverified PRs, and Google reviewers spend time back 
+and forth, fixing the PR. Sometimes PR is just not moving forward because of the
+lengthy feedback loop.
+
+With the new standalone Keras repository, external contributors
+should experience much shorter turn-around time when 
+building/testing Keras, since they don't need to build TensorFlow anymore.
+This should  have a positive impact on building a vibrant open-source
+developer community.
+
+In addition, by getting the Keras team at Google to start developing Keras
+using the same public tools and infrastructure as third-party developers,
+we make the development process more transparent and more community-oriented.
+
+### TensorFlow API modularity
```

Comment:
That to me is the main motivation which could justify splitting the repositories (the build time one just suggests we should split the continuous integration tooling, not the repositories).

Reply:
Done. Updated.